views.go

```diff
@@ -0,0 +1,221 @@
+package slack
+
+import (
+	"context"
+	"encoding/json"
+)
+
+const (
+	VTModal   ViewType = "modal"
+	VTHomeTab ViewType = "home"
+)
+
+type ViewType string
+
+type View struct {
```

Comment:
All of the fields in this struct should be `omitempty` with the exception of `Type`, `Title`, and `Blocks` which are required View payload fields.
https://api.slack.com/reference/surfaces/views

This should allow usage of the View struct in push/update View API calls

Reply:
For the sake of avoiding excessive implementation sharing, I created `ModalViewRequest` and `HomeTabViewRequest` indivisually.
And then the role of `View` is an entity in response from Slack.

What do you think?

Reply:
Ah okay that makes sense. Well in that case, wouldn't we still wan't to omit-empty the non-required fields on the `ModalViewRequest` and `HomeTabViewRequest` structs? It might not be necessary, but I have noticed that the Slack API can be extremely picky about included fields in a payload, e.g. leaving `CallbackID` unset will actually send the Slack API `"callback_id": ""`, which could cause it to blow up, where as just omitting the field makes it happy as a clam.

Reply:
Ahh I didn't see that, that makes perfect sense. Thank you.